runtime/libc/cortex_m_backing.c

```diff
@@ -4,6 +4,41 @@ int printf_(const char* format, ...);
 
 volatile int CORTEX_M_ARG_C = 0;
 
+int runtime_main(int argc, char** argv) {
+    // Setup the linear memory and function table
+    alloc_linear_memory();
+    populate_table();
+    // Setup our allocation logic
+    runtime_heap_base = wasmg___heap_base;
+    printf("starting rhb %d\n", runtime_heap_base);
+    if (runtime_heap_base == 0) {
+        runtime_heap_base = memory_size;
+    }
+    // Setup the global values (if needed), and populate the linear memory
+    switch_out_of_runtime();
+    populate_globals();
+    switch_into_runtime();
+    populate_memory();
```

Comment:
Let's create a function for this. I think if theory 
```
    // Setup the linear memory and function table
    alloc_linear_memory();
    populate_table();
    switch_out_of_runtime();
    populate_globals();
    switch_into_runtime();
    populate_memory();
```
Could all run at the start of the function